ush/remap.sh

```diff
@@ -0,0 +1,327 @@
+	varname_update "yr" "${yvar}" "${var_rotate_path}"
+	cdo_remap "${yvar}" "${var_rotate_path}" "${interp_type}" "${output_path}"
+
+    else
+        echo "FATAL ERROR: Vector rotations with ${nangles} attributes is not supported. Aborting!!!"
+        exit 102
+    fi
+    
+    echo "Updating variable arrays following vector rotation."
+    rm -f "${var_rotate_path}" >> /dev/null
+}
+
+#######
+
+# varname_update - Rename a variable in a netCDF file and write to a new file.
+#
+# Description: This function renames a variable in a netCDF file from
+#   `old_varname` to `new_varname` and writes the result to the
+#   (existing) netCDF file specified by `ncfile`.
+#
+# Parameters:
+#   $1 - The old variable name to be renamed.
+#   $2 - The new variable name to use.
+#   $3 - The path to the input netCDF file; note that this is updated
+#        with every call to this function.
+#
+# Example usage:
+#   varname_update "old_variable" "new_variable" "input_file.nc"
+#
+# This example renames the variable "old_variable" to "new_variable"
+# in the input netCDF file and writes the result to the specified
+# netCDF file.
+function varname_update(){
+    local old_varname="${1}"
+    local new_varname="${2}"
+    local ncfile="${3}"
+
+    echo "Renaming variable ${old_varname} to ${new_varname} and writing to file ${ncfile}."
+    ncrename -O -v "${old_varname}","${new_varname}" "${ncfile}" >> "${REMAP_LOG}" 2>&1
+}
+
+#######
+
+# Clobber the output file; otherwise the respective variables will be
+# written multiple times to the file; for example VAR, VAR_1, VAR_2,
+# etc.,
+rm -f "${output_path}" >& /dev/null
+
+# Define a local file path to be used for file manipulations.
+#workgrid_path="${PWD}/workgrid.nc"
+
+# Read the configuration file for the the variables to be remapped and
+# proceed accordingly.
+while IFS= read -r line; do
+
+    # Get the attributes for the respective variable(s).
+    varname=$(awk '{print $1}' <<< "${line}")
+    interp_type=$(awk '{print $2}' <<< "${line}")
+    srcgrid=$(awk '{print $3}' <<< "${line}")
+    rotate=$(awk '{print $4}' <<< "${line}")
+    angle=$(awk '{print $5}' <<< "${line}")
+    
+    if (( rotate == 0 )); then
+        # No rotation necessary; interpolate/remap the variables and
+        # directly.
+        echo "Remapping variable ${varname} without rotation."
+        cdo_remap "${varname}" "${srcgrid}" "${interp_type}" "${output_path}"
+	
+    elif (( rotate == 1 )); then
+        # Rotation necessary; rotate the respective vector quantities
+        # relative to the source grid projection and subsequently
+        # remap the variables to the specified destination grid.
+        echo "Remapping variables ${varname} with rotation."
+        cdo_rotate "${varname}" "${srcgrid}" "${interp_type}" "${angle}"
+	
+    else
+        echo "FATAL ERROR: Rotation option ${rotate} not recognized. Aborting!!!"
+        exit 101
+    fi
+    
+done < "${variable_file}"
+
```

Comment:
To fit into the rest of the global workflow, it is suggested that the final remapped netcdf output follows the format of the following NetCDF file on Hera in terms of the global attributes, variable attributes, and coordinate names: `/scratch2/NCEPDEV/ensemble/noscrub/Eric.Sinsky/MOM6/ocnr2013100106.1.2013100100_0p25x0p25_MOM6.nc`.  This output is needed as input for the reg2grb2 program.

Reply:
@EricSinsky-NOAA OK, this is going to need to be a step after application of `remap.sh`. This application is agnostic to the forecast model. Similar to PR #1875, another step will be need to update the attributes accordingly.

Reply:
Thank you, @HenryWinterbottom-NOAA. If there is any more information you need that could help, please let me know.